pkg/xds/server.go

```diff
@@ -91,10 +92,21 @@ type WatchedResource struct {
 	LastError string
 
 	// LastResources tracks the contents of the last push.
-	// This field is extremely expensive to maintain and is typically disabled
+	// This field is extremely expensive to maintain and is typically disabled.
+	// It is only used for testing now.
 	LastResources Resources
 }
 
+func (r *WatchedResource) DeepCopy() *WatchedResource {
+	out := *r
+	if len(r.ResourceNames) > 0 {
```

Comment:
I am worried about this cost. Especially since its to support a debug interface, which doesn't even depend on resourceNames